web-app/django/VIM/apps/main/utils/email.py

```diff
@@ -0,0 +1,109 @@
+"""
+Email utilities for user verification and notifications.
+Provides async email sending and token generation for email verification.
+"""
+
+import threading
+from django.core.mail import EmailMessage
+from django.contrib.auth.tokens import default_token_generator
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from django.utils.encoding import force_bytes, force_str
+from django.template.loader import render_to_string
+from django.conf import settings
+from django.contrib.auth.models import User
+
+
+def send_email_async(subject, body_html, to_emails):
+    """
+    Send HTML email asynchronously using threading to avoid blocking the request.
+
+    Args:
+        subject: Email subject line
+        body_html: HTML content of the email
+        to_emails: List of recipient email addresses
+    """
+
+    def send_email_task():
+        email = EmailMessage(
+            subject=subject,
+            body=body_html,
+            from_email=settings.DEFAULT_FROM_EMAIL,
+            to=to_emails,
+        )
+        email.content_subtype = "html"
+        email.send()
```

Comment:
For cleanliness, I would define this function outside of this one and then call in `threading.Thread` with the `args` or `kwargs` arguments for `subject`, `body_html` and `to_emails` arguments...

Reply:
For the future, we might also want to return something from this function and await it somewhere so that a success or error can be displayed.